### DISCUSSION TITLE
crazyflie-lib-python send_zdistance_setpoint

### DISCUSSION THREAD

**Discussion by 7enHom** [bitcraze/discussions#1151] [Q&A] [answered]
Hi!

I'm trying to send the setpoint for angular rate, and I know that the 'flightmode.stabModeRoll' and 'flightmode.stabModePitch' params should be set to '0' when using commander.send_setpoint(). But when I try to use commander.send_zdistance_setpoint() to send the angular rate, it doesn't work anymore. I just want to know if this function can set the angular rate?

Thanks！

**Comment by knmcguire** [comment, marked as answer]
Hi! I think that the decoder for that packet (zdistance hold with attitude control) is here in the firmware: https://github.com/bitcraze/crazyflie-firmware/blob/d1339b8bde821be6b425859e89595b4e70329924/src/modules/src/crtp_commander_generic.c#L126. As you can see, the roll pitch and yaw mode are hardcoded in there,  unlike the send_setpoint decoder that exists in this function: https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/crtp_commander_rpyt.c. 

So no, that is not possible to change it to angular rate and you'll need to change the decoder on the firmware side of that.

**Reply by 7enHom** [reply]
Thanks a lot!
I changed the decoder on the firmware side of zdistance and it works!

**Reply by knmcguire** [reply]
ah awesome! 

(could you select 'mark as answer' then 😄 ?)
